src/doc/CMakeLists.txt

```diff
@@ -56,14 +60,21 @@ endif()
 # HTML output directory
 set(SPHINX_HTML_DIR "${CMAKE_CURRENT_BINARY_DIR}/html")
  
-configure_file(
-    "${CMAKE_CURRENT_SOURCE_DIR}/conf.py"
-    "${CMAKE_CURRENT_BINARY_DIR}/conf.py"
-    @ONLY)
- 
+if(WIN32)
+    set(sphinx_launcher ${PYTHON_EXECUTABLE}) 
+endif()
+
 add_custom_target(gui_manual ALL
+    ${sphinx_launcher}
     ${SPHINX_EXECUTABLE}
         -a -q -b html
+        -D html_theme_path=${SPHINX_THEME_DIR}
```

Comment:
Use of python to launch sphinx-build is necessary on my win32 system, the way sphinx is installed, or some such. Doesn't seem necessary on linux, so setting of the sphinx_launcher variable IS WIN32 specific.  That is the only code in the WIN32 if-def.

I decided to have as few mods in the conf.py file as possible.  Passing the necessary overrides for them, etc in the arguments to sphinx-build seems the way to go.

BTW, the previous version of the CMakeLists.txt file would not create themed html even on linux.  And it called 'configure_file' on conf.py that DID NOTHING.  Not only that, but sphinx-build was parsing the conf.py in source directory anyways, so it seemed unnecssary to put a copy in the build dir.

Reply:
I guess I am thinking the opposite...keeping ``CMakeLists.txt`` file as free from sphinx-build related stuff as possible and handling as much as possible regarding sphinx-build in ``conf.py``. Why? Its two places to have to manage sphinx-build behavior instead of one. What I am suggesting is putting a block like the following *in* ``conf.py`` file

```
if not os.environ.get('READTHEDOCS'):
    html_theme = sphinx_rtd_theme
    html_theme_dir = _themes
    version = "local build"
```

I am punting on how version info gets in so not quite meeting your goals here.  But, don't think having explicit version info for a local build is all that important either.

Above elims most changes to ``CMakeLists.txt`` and all those having to do with sphinx-build *behavior*. Having the actual make target is IMHO only marginally useful at that point too but understand that invokation on Windows is different enough it should probably be handled there.

Does this work?